packages/streamline/src/components/inputs/input-text/components/input-text-icon.tsx

```diff
@@ -0,0 +1,107 @@
+import { IconsName } from '../../../../primitives/icon/icon.types';
+import React from 'react';
+import { ViewProps, StyleProp, ViewStyle } from 'react-native';
+import { Icon } from '../../../../primitives/icon/icon';
+import { Size } from '../../../../primitives/icon/icon.types';
+import { ColorTheme } from '../../../../theme';
+import { Box } from '../../../../primitives/box/box';
+
+type StyleContextType = {
+  isTextInputFocused: boolean;
+  forceFocus: () => void;
+  disabled?: boolean;
+};
+
+const StyleContext = React.createContext<StyleContextType>({
+  isTextInputFocused: false,
+  forceFocus: () => undefined,
+});
+
+export const IconAdornment: React.FunctionComponent<
+  {
+    icon: React.ReactNode;
+  } & Omit<StyleContextType, 'style'>
+> = ({ icon, isTextInputFocused, forceFocus, disabled }) => {
+  const contextState = { isTextInputFocused, forceFocus, disabled };
+
+  return (
+    <StyleContext.Provider value={contextState}>{icon}</StyleContext.Provider>
+  );
+};
+
+interface Props {
+  name: IconsName;
+  disabled?: boolean;
+  loading?: boolean;
+  forceTextInputFocus?: boolean;
+  onPress?: () => void;
+  pointerEvents?: ViewProps['pointerEvents'];
+}
+
+const getConfig = ({
+  disabled,
+  loading,
+  name,
+}: {
+  disabled?: boolean;
+  loading: boolean;
+  name: string;
+}) => {
+  let color: ColorTheme = 'BLUKO_500';
+  let size: Size = 'regular';
+  let extraStyle: StyleProp<ViewStyle>;
+
+  if (disabled) {
+    color = 'GREY_500';
+  }
+  if (name === 'ChevronDown') {
+    color = 'GREY_400';
+    size = 'regular';
+    extraStyle = { paddingTop: 4 };
+  }
+  if (name === 'ChevronDown' && disabled) {
+    color = 'GREY_300';
+  }
+  if (loading) {
+    color = 'BLUKO_500';
+    size = 'regular';
+  }
+  return { color, size, extraStyle };
```

Comment:
I find using let when not necessary possibly leading to confusion as when you try to find the return value, you need to take into account several possible assignations. I would rather see something like this : 

```tsx
if (loading) {
  return {
    color: 'BLUKO_500',
    size: 'regular',
  }
}

if (name === 'ChevronDown') {
  [...]
}

if (disabled) {
  [...]
}

return {
  color: 'BLUKO_500',
  size: 'regular',
}
```

If it becomes too complex, maybe that's a sign that sub-methods could be nice.

Reply:
Isn't it always 'regular' size?